base_rest/controllers/main.py

```diff
@@ -44,48 +44,81 @@ def __init__(cls, name, bases, attrs):  # noqa: B902
         root_path = getattr(cls, "_root_path", None)
         collection_name = getattr(cls, "_collection_name", None)
         if root_path and collection_name:
-            if not hasattr(cls, "_module"):
-                cls._module = _get_addon_name(cls.__module__)
+            cls._module = _get_addon_name(cls.__module__)
             _rest_controllers_per_module[cls._module].append(
                 {
                     "root_path": root_path,
                     "collection_name": collection_name,
                     "controller_class": cls,
                 }
             )
+            _logger.info(
+                "Added rest controller %s for module %s",
+                _rest_controllers_per_module[cls._module][-1],
+                cls._module,
+            )
 
 
 class RestController(Controller, metaclass=RestControllerType):
     """Generic REST Controller
 
-    This controller provides generic routes conform to commen REST usages.
-    You must inherit of this controller into your code to register your REST
-    routes. At the same time you must fill 2 required informations:
+    This controller is the base controller used by as base controller for all the REST
+    controller generated from the service components.
+
+    You must inherit of this controller into your code to register the root path
+    used to serve all the services defined for the given the collection name.
+    your REST. This registration requires 2 parameters:
 
     _root_path:
     _collection_name:
 
+    Only one controller by _collection_name, _root_patch should exists into an
+    odoo database. If more than one controller exists, a warning is issued into
+    the log at startup and the concrete controller used as base class
+    for the services registered into the collection name and served at the
+    root path is not predictable.
```

Comment:
this is not clear, can you rephrase?

Reply:
Is-it clearer now? (btw thank you for the review)